framework/include/material_image_helpers.hpp

```diff
@@ -354,6 +354,31 @@ namespace gvk
 		add_tuple_or_indices(aResult, rest...);
 	}
 
+	template <typename... Rest>
+	void add_tuple_or_indices_shared(std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aResult)
+	{ }
+
+	template <typename... Rest>
+	void add_tuple_or_indices_shared(std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aResult, model& aModel, Rest&... rest)
+	{
+		aResult.emplace_back(avk::shared(aModel), std::vector<size_t>{});
+		add_tuple_or_indices_shared(aResult, rest...);
+	}
+
+	template <typename... Rest>
+	void add_tuple_or_indices_shared(std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aResult, size_t aMeshIndex, Rest&... rest)
+	{
+		std::get<std::vector<size_t>>(aResult.back()).emplace_back(aMeshIndex);
+		add_tuple_or_indices_shared(aResult, rest...);
+	}
+
+	template <typename... Rest>
+	void add_tuple_or_indices_shared(std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aResult, std::vector<size_t> aMeshIndices, Rest&... rest)
```

Comment:
```suggestion
void add_tuple_or_indices_shared(std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aResult, const std::vector<size_t>& aMeshIndices, Rest&... rest)
```

`const vector&` should be more efficient here. It is only read in the body => no point in passing it by  value?!

Reply:
^ ah, probably needs to be `vector&` (i.e. non-const) to be compatible with `Rest&...`!